packages/core/src/lib/markdown-it/index.js

```diff
@@ -75,31 +75,77 @@ markdownIt.renderer.rules.fence = (tokens, idx, options, env, slf) => {
   if (highlightLinesInput) {
     const highlightLines = highlightLinesInput.split(HIGHLIGHT_LINES_DELIMITER_REGEX);
     highlightRules = highlightLines
-      .map(ruleStr => HighlightRule.parseRule(ruleStr, -startFromZeroBased, lines))
+      .map(ruleStr => HighlightRule.parseRule(ruleStr, -startFromZeroBased, str.split('\n')))
       .filter(rule => rule); // discards invalid rules
   }
 
   if (lang && hljs.getLanguage(lang)) {
     try {
-      /* We cannot syntax highlight THEN split by lines. For eg:
+      /* With highlightjs version >= v10.7.0, usage of continuation is deprecated
+
+      For the purposes of line-by-line highlighting, we have to first highlight the
+      whole block, then split the resulting html string according to '\n', and add
+      the corresponding opening and closing tags for the html string to be well-formed and
+      maintain the correct state per line.
+      eg:
       ```markdown
       *****
       -----
       ```
 
       becomes
 
-      <span class="hljs-section">*****
-      -----</span>
-      Note the line break contained inside a <span> element.
-      So we have to split by lines THEN syntax highlight.
+      "<span class="hljs-section"><span class="hljs-strong">****</span>
+      <span class="hljs-emphasis">*\n-----\n</span></span>"
+
+      after splitting on '\n':
+
+      ["<span class="hljs-strong">****</span><span class="hljs-emphasis">*",
+       "-----",
+       "</span>",
+      ]
+
+      tokenStack maintains the visited tokens. For eg, for <span class="xyz">, the token would be "xyz".
+      If it encounters a </span>, it pops from the stack.
+      Else, it pushes the token to the stack.
+
+      At the end of each line, it will be prepended by the necessary
+      opening tags and appended by the neccessary closing tags.
```

Comment:
The comment becomes a bit too long. Maybe can trim out this part and refer to the GitHub issue you're basing the approach on (or even the StackOverflow link in the discussion of that issue). Imo the comment should just give an example, the result after calling `highlight`, and the result after you post-process it.

(Aside, this could be a good design notes writeup)